contracts/src/v0.8/dev/VRFSubscriptionBalanceMonitor.sol

```diff
@@ -0,0 +1,314 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+import "../ConfirmedOwner.sol";
+import "../interfaces/KeeperCompatibleInterface.sol";
+import "../interfaces/VRFCoordinatorV2Interface.sol";
+import "../interfaces/LinkTokenInterface.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+
+/**
+ * @title The VRFSubscriptionBalanceMonitor contract.
+ * @notice A keeper-compatible contract that monitors and funds VRF subscriptions.
+ */
+contract VRFSubscriptionBalanceMonitor is ConfirmedOwner, Pausable, KeeperCompatibleInterface {
+  VRFCoordinatorV2Interface public COORDINATOR;
+  LinkTokenInterface public LINKTOKEN;
+
+  uint256 private constant MIN_GAS_FOR_TRANSFER = 55_000;
+
+  event FundsAdded(uint256 amountAdded, uint256 newBalance, address sender);
+  event FundsWithdrawn(uint256 amountWithdrawn, address payee);
+  event TopUpSucceeded(uint64 indexed subscriptionId);
+  event TopUpFailed(uint64 indexed subscriptionId);
+  event KeeperRegistryAddressUpdated(address oldAddress, address newAddress);
+  event VRFCoordinatorV2AddressUpdated(address oldAddress, address newAddress);
+  event LinkTokenAddressUpdated(address oldAddress, address newAddress);
+  event MinWaitPeriodUpdated(uint256 oldMinWaitPeriod, uint256 newMinWaitPeriod);
+
+  error InvalidWatchList();
+  error OnlyKeeperRegistry();
+  error DuplicateSubcriptionId(uint64 duplicate);
+
+  struct Target {
+    bool isActive;
+    uint96 minBalanceJuels;
+    uint96 topUpAmountJuels;
+    uint56 lastTopUpTimestamp;
+  }
+
+  address private s_keeperRegistryAddress;
```

Comment:
Any reason keeper registry is private as opposed to the public VRF coordinator?

Reply:
No significant reason, it's like that in ERC20BalanceMonitor as well.